scripts/rnnlm/validate_data_dir.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(description="Validates data directory containing text "
+                                 "files from one or more data sources, including dev.txt.",
+                                 epilog="E.g. " + sys.argv[0] + " data/rnnlm/data",
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+parser.add_argument("--spot-check", type=str, default='true',
+                    choices=['true', 'false'],
+                    help="If true, only do spot check on text files.")
+parser.add_argument("--allow-internal-eos", type=str, default='true',
+                    choices=['true', 'false'],
+                    help="If true, allow internal </s> in lines of the text.")
+parser.add_argument("data_dir",
+                    help="Directory in which to look for text data")
+
+args = parser.parse_args()
+
+EOS_SYMBOL = '</s>'
+SPECIAL_SYMBOLS = ['<s>', '<brk>', '<eps>']
+
+if not os.path.exists(args.data_dir):
+    sys.exit(sys.argv[0] + ": Expected directory {0} to exist".format(args.data_dir))
+
+if not os.path.exists("{0}/dev.txt".format(args.data_dir)):
+    sys.exit(sys.argv[0] + ": Expected file {0}/dev.txt to exist".format(args.data_dir))
+
+
+num_text_files = 0
+
+
+def check_text_file(text_file):
+    with open(text_file, 'r', encoding="utf-8") as f:
+        found_nonempty_line = False
+        lineno = 0
+        if args.allow_internal_eos == 'true':
+            disallowed_symbols = SPECIAL_SYMBOLS
+        else:
+            disallowed_symbols = SPECIAL_SYMBOLS + [EOS_SYMBOL]
+        for line in f:
+            line = line.strip("\n")
+            if line is None:
+                break
+            lineno += 1
+            if args.spot_check == 'true' and lineno > 10:
+                break
+            words = line.split()
+            if len(words) != 0:
+                found_nonempty_line = True
+                for word in words:
+                    if word.lower() in disallowed_symbols:
```

Comment:
Let's not do 'lower()' here; I'd rather allow the upper-case versions of those symbols as real words since in general the whole system will be case-sensitive.